cli/js/colors.ts

```diff
@@ -1,77 +0,0 @@
-// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
-
-// TODO(kitsonk) Replace with `deno_std/colors/mod.ts` when we can load modules
-// which end in `.ts`.
-
-import { noColor } from "./deno.ts";
-
-interface Code {
-  open: string;
-  close: string;
-  regexp: RegExp;
-}
-
-const enabled = !noColor;
-
-function code(open: number, close: number): Code {
-  return {
-    open: `\x1b[${open}m`,
-    close: `\x1b[${close}m`,
-    regexp: new RegExp(`\\x1b\\[${close}m`, "g"),
-  };
-}
-
-function run(str: string, code: Code): string {
-  return enabled
-    ? `${code.open}${str.replace(code.regexp, code.open)}${code.close}`
-    : str;
-}
-
-export function bold(str: string): string {
-  return run(str, code(1, 22));
-}
-
-export function italic(str: string): string {
-  return run(str, code(3, 23));
-}
-
-export function yellow(str: string): string {
-  return run(str, code(33, 39));
-}
-
-export function cyan(str: string): string {
-  return run(str, code(36, 39));
-}
-
-export function red(str: string): string {
-  return run(str, code(31, 39));
-}
-
-export function green(str: string): string {
-  return run(str, code(32, 39));
-}
-
-export function bgRed(str: string): string {
-  return run(str, code(41, 49));
-}
-
-export function white(str: string): string {
-  return run(str, code(37, 39));
-}
-
-export function gray(str: string): string {
-  return run(str, code(90, 39));
-}
-
-// https://github.com/chalk/ansi-regex/blob/2b56fb0c7a07108e5b54241e8faec160d393aedb/index.js
-const ANSI_PATTERN = new RegExp(
-  [
-    "[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?\\u0007)",
-    "(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-ntqry=><~]))",
-  ].join("|"),
-  "g"
-);
-
-export function stripColor(string: string): string {
-  return string.replace(ANSI_PATTERN, "");
-}
```

Comment:
This is the ANSI_PATTERN that didn't work, and it was in the deprecated colors.ts file. I tried to use it by copying it into fmt/colors.ts, and I replaced it with a new, shorter and working one.

Reply:
do you have an example of a string (with ansi escape codes) for which it doesn't work?

Reply:
... I don't know what happened when I was coding, but I am testing it and... it works. :/
Do you think we should bring back the previous regex? This one is at least shorter.

Reply:
```js
const ANSI_PATTERN = new RegExp(
  [
    "[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?\\u0007)",
    "(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-ntqry=><~]))",
  ].join("|"),
  "g"
);

const NEW_ANSI_PATTERN = /(\x9B|\x1B\[)[0-?]*[ -\/]*[@-~]/gi

console.log(rgb24(bold(dim(italic(underline("HELLO")))), 0xff3280))
console.log(rgb24(bold(dim(italic(underline("HELLO")))), 0xff3280).replace(ANSI_PATTERN, ""))
console.log(rgb24(bold(dim(italic(underline("HELLO")))), 0xff3280).replace(NEW_ANSI_PATTERN, ""))
```

This code produces the following output:
![image](https://user-images.githubusercontent.com/13885008/81975648-a424fd80-9627-11ea-8513-34717a6de720.png)

Reply:
Cool - these are good to have. Can you evaluate `rgb24(bold(dim(italic(underline("HELLO"))))` these and provide the exact escape code sequences? We should use these as test cases for strip_ansi_colors in Rust.